consensus/pow/engine.go

```diff
@@ -144,3 +144,28 @@ func verifyTarget(header *types.BlockHeader) error {
 func getMiningTarget(difficulty *big.Int) *big.Int {
 	return new(big.Int).Div(maxUint256, difficulty)
 }
+
+// returns the second mining target based on the difference of current clock time
+// and the timestamp of parent block. Initially, the difficulty should be high
+// This function is used to determine which coinbase can mine.
+
+func getSecondMiningTarget(time uint64, parentHeader *types.BlockHeader) *big.Int {
```

Comment:
Is it designed to resolve the sharding security issue? If yes, please discuss with other team members firstly, we should ensure the solution is doable before any code merge.